released/packages/coq-kruskal-higman/coq-kruskal-higman.1.3/opam

```diff
@@ -0,0 +1,41 @@
+opam-version: "2.0"
+synopsis: "Extending Coq library for manipulating Almost Full relations with Higman's lemma"
+description: """
+   This library formalizes additional tools for AF relations, eg quasi morphisms applied to Higman's lemma.
+"""  
+maintainer: ["Dominique Larchey-Wendling (https://github.com/DmxLarchey)"] 
+authors: "Dominique Larchey-Wendling (https://github.com/DmxLarchey)"
+license: "MPL-2.0"
+homepage: "https://github.com/DmxLarchey/Kruskal-Higman/"
+bug-reports: "https://github.com/DmxLarchey/Kruskal-Higman/issues"
+dev-repo: "git+https://github.com:DmxLarchey/Kruskal-Higman/"
+
+build: [
+  [make "-j%{jobs}%" "type"]
+]
+install: [
+  [make "install"]
+]
+
+depends: [
+  "coq-kruskal-trees" {}
+  "coq-kruskal-finite" {}
+  "coq-kruskal-almostfull" {}
+  "coq-kruskal-fan" {}
```

Comment:
@DmxLarchey do you really want these dependencies to be version-free? For example, are you really sure this works with `coq-kruskal-trees.1.0` from two years ago?

Reply:
Well `coq-kruskal-higman` depends on `coq-kruskal-fan` which has stronger versions dependencies. I did list the packages in here, even though `coq-kruskal-fan` would be enough,  because it corresponds to `Import` that actually occur in the code of `coq-kruskal-higman`. 

Is there a policy for what should be listed as dependencies wrt transitive dependencies?

Regarding your specific question, I am pretty sure that the code higman itself would work with old versions of trees if it were not excluded by the requirements of fan. Actually, fan was initially part of higman in older versions of the project.

Reply:
Note that, if `coq-kruskal-higman` explicitly requires files from `coq-kruskal-fan` only, then I would suggest that future releases only use the latter as a dependency.

But Karl's original concern is still valid. Is `coq-kruskal-higman` compatible with any version of `coq-kruskal-fan`? Sure, the latter might have strong version dependencies. But since there is no strong version dependency on `coq-kruskal-fan`, that is the same as not having any. (More precisely, that is the same as having the union of all the version dependencies of `coq-kruskal-fan`.)

For example, `coq-kruskal-higman` 1.2 requires `coq-kruskal-fan` >= 1.1. But `coq-kruskal-higman` 1.3 allows `coq-kruskal-fan` 1.0. This looks very dubious.

Reply:
Thx @silene for the input. But the direct dependencies of `coq-kruskal-higman` in terms of the `Import` ed modules are those listed indeed which actually covers the whole dependency trees (except for `coq`). As for versions, it might be a surprise but the external interface did not change much (or at all), explaining why the new revision of `coq-kruskal-higman` could still work with older revision of its dependencies. 

But upgrading to `coq 8.20` imposed an update because I did include a version dependency on `coq` in nearly all opam files of `coq-kruskal-*` at first. A mistake I think in retrospect.

For instance if I add a dependency on `coq` for versions `< 8.21` to `coq-kruskal-*`, it will mean that when `8.21` pops up, then I need to publish an update all the `coq-kruskal-*` opam files even if no code update is necessary. Which happened for most of these packages regarding `8.20`.

Reply:
@DmxLarchey nothing stops you from just having a lower bound on Coq, such as:
```
"coq" {>= "8.18"}
```